### PR TITLE
drivers: espi: npcx: support espi taf rpmc request handler

### PR DESCRIPTION
**Pull request by TomChang19**
This commit adds support for handling eSPI TAF RPMC OP1 and OP2 requests.

### CODE REVIEW

File: drivers/espi/espi_taf_npcx.c [hunk 1]
```diff
@@ -444,6 +452,69 @@ static int espi_taf_npcx_flash_erase(const struct device *dev, struct espi_saf_p
 	return 0;
 }
 
+#ifdef CONFIG_ESPI_TAF_NPCX_RPMC_SUPPORT
+static int espi_taf_npcx_rpmc_op1(const struct device *dev, struct espi_saf_packet *pckt)
+{
+	struct espi_taf_npcx_pckt *taf_data_ptr = (struct espi_taf_npcx_pckt *)pckt->buf;
+	uint8_t *data_ptr = (uint8_t *)(taf_data_ptr->data);
```
Comment (fabiobaltieri) [inline drivers/espi/espi_taf_npcx.c:459]: isn't data a `uint8_t *` alrady?

```c
struct espi_taf_npcx_pckt {
        uint8_t tag;
        uint8_t *data;
};  
````

File: drivers/espi/espi_taf_npcx.c [hunk 2]
```diff
@@ -444,6 +452,69 @@ static int espi_taf_npcx_flash_erase(const struct device *dev, struct espi_saf_p
...
+{
+	struct espi_taf_npcx_pckt *taf_data_ptr = (struct espi_taf_npcx_pckt *)pckt->buf;
+	uint8_t dummy_byte = 0;
+	struct npcx_ex_ops_uma_in op_in = {
+		.opcode = ESPI_TAF_RPMC_OP2_CMD,
+		.tx_buf = &dummy_byte,
+		.tx_count = 1,
+		.rx_count = pckt->len,
+	};
+	uint8_t sts_buf[64];
```
Comment (fabiobaltieri) [inline drivers/espi/espi_taf_npcx.c:494]: oof, quite a big buffer to declare on the stack, anyway, is that 64 fixed? should there be a #define for it?

File: drivers/espi/espi_taf_npcx.c [hunk 3]
```diff
@@ -444,6 +452,69 @@ static int espi_taf_npcx_flash_erase(const struct device *dev, struct espi_saf_p
...
+	int rc;
+
+	rc = flash_ex_op(spi_dev, FLASH_NPCX_EX_OP_EXEC_UMA, (uintptr_t)&op_in, &op_out);
+	if (rc) {
+		LOG_ERR("flash RPMC OP2 fail");
+		return -EIO;
+	}
+
+	rc = taf_npcx_completion_handler(dev, CYC_SCS_CMP_WITH_DATA_ONLY, taf_data_ptr->tag,
+					 pckt->len, (uint32_t *)&sts_buf[0]);
```
Comment (fabiobaltieri) [inline drivers/espi/espi_taf_npcx.c:508]: You could also just do `(uint32_t *)sts_buf` up to you really, referencing and de referencing 0 seems redundant, same above (iine 496)

also: is there anything to block this from overrunning the buffer? Would have expected to see a check against `sizeof(sts_buf)` somewhere

File: drivers/espi/espi_taf_npcx.c (hunk 1 above)
Comment (TomChang19) [reply drivers/espi/espi_taf_npcx.c]: Done. The data is already a ```uint8_t *```. Have removed this casting.

File: drivers/espi/espi_taf_npcx.c (hunk 3 above)
Comment (TomChang19) [reply drivers/espi/espi_taf_npcx.c]: Done. Removed redundant content and added a conditional statement to check the length.

File: drivers/espi/espi_taf_npcx.c (hunk 2 above)
Comment (TomChang19) [reply drivers/espi/espi_taf_npcx.c]: The maximum payload size is 64 bytes. This buffer is used to save the response data from Flash. Add the definition for maximum payload size.

**Review by fabiobaltieri — APPROVED** [review]

**Review by albertofloyd — APPROVED** [review]

**Pull request merged**